src/Avalonia.Controls/Presenters/TextPresenter.cs

```diff
@@ -204,6 +213,11 @@ internal void CaretIndexChanged(int caretIndex)
 
         protected override FormattedText CreateFormattedText(Size constraint)
         {
+            if (PasswordChar != default(char))
+            {
+                Text = new string(PasswordChar, Text.Length);
```

Comment:
You don't want to set the `Text` property here. This is two-way bound with `TextBox.Text`! If you look in the devtools you'll see that `TextBox` now has "*****" as its text. This isn't what we want. Just use a local  variable.